src/Cameras/VR/babylon.vrExperienceHelper.ts

```diff
@@ -309,8 +304,8 @@ module BABYLON {
 
             // Gamepad connection events
             this._webVRCamera.onControllerMeshLoadedObservable.add((webVRController) => this._onDefaultMeshLoaded(webVRController));
-            this._scene.gamepadManager.onGamepadConnectedObservable.add((pad) => this._onNewGamepadConnected(pad));
-            this._scene.gamepadManager.onGamepadDisconnectedObservable.add((pad) => this._onNewGamepadDisconnected(pad));
+            this._scene.gamepadManager.onGamepadConnectedObservable.add(this._onNewGamepadConnected);

```

Comment:
you are losing the this context here no?

Reply:
oh wait it should work as I've just seen the change on onNewGamepadConnected to arrow function

Reply:
did you tested it with controllers and gamepads and so on?

Reply:
Yes, tested disposing with and without controllers in and made sure sponza and mansion still work as expected.